src/map/npc.cpp

```diff
@@ -5089,6 +5089,17 @@ static const char* npc_parse_mob(char* w1, char* w2, char* w3, char* w4, const c
 		return strchr(start,'\n');// skip and continue
 	}
 
+	// Check if sprite is the mob name or ID
+	char *pid;
+	mob_id = strtol(sprite, &pid, 0);
+
+	if (pid != nullptr && *pid != '\0') {
+		std::shared_ptr<s_mob_db> mob = mobdb_search_aegisname(sprite);
+
+		if (mob != nullptr)
+			mob_id = mob->id;
```

Comment:
Should be inverted and an error message with %s should be used, the other one is %d.